### PR TITLE
server: ignore ClientPluginAuthLenencClientData temporarily

### PR DESCRIPTION
**Pull request by tiancaiamao**
This is a hot fix, we need investigate it for more detail
https://github.com/pingcap/tidb/issues/1768

### CODE REVIEW

**Comment by shenli** [comment]
Please add a test case.

**Comment by shenli** [comment]
We could follow the logic in mysql: https://github.com/mysql/mysql-server/blob/e0e0ae2ea27c9bb76577664845507ef224d362e4/sql-common/client.c#L3694

**Comment by tiancaiamao** [comment]
I think we should do it in a real fix, not here, this PR is just a hot fix if necessary.

we should find why it fail, what's be best way to solve it;
if mysql client doesn't follow its document, we should feed back an issue;
we should also think how to make our code more robust to handle incorrect behavior, including test cases to cover that.

**Comment by shenli** [comment]
I mean any PR should prove its correctness. At lease you should add a test case.

**Comment by tiancaiamao** [comment]
According to [document](http://dev.mysql.com/doc/internals/en/capability-negotiation.html) , 

> The client should only announce the capabilities in the Handshake Response Packet that it has in common with the server.

It's a bug of mysql client, here it is: https://github.com/mysql/mysql-server/blob/5.7/sql-common/client.c#L3478

After `~(CLIENT_COMPRESS | CLIENT_SSL | CLIENT_PROTOCOL_41) | mysql->server_capabilities`, it turn on many bits to 0x1 that should not.

**Comment by tiancaiamao** [comment]
PTAL @shenli

**Comment by shenli** [comment]
LGTM
@tiancaiamao @zimulala PTAL

File: server/conn.go [hunk 1]
```diff
@@ -193,9 +193,13 @@ func handshakeResponseFromData(packet *handshakeResponse41, data []byte) error {
 	pos += len(packet.User) + 1
 
 	if capability&mysql.ClientPluginAuthLenencClientData > 0 {
-		// TODO: Support mysql.ClientPluginAuthLenencClientData, skip it now
+		// MySQL client set the wrong capability, it will set this bit even server doesn't
```
Comment (zimulala) [inline server/conn.go]: s/set/sets

**Comment by zimulala** [comment]
LGTM

**Pull request merged**